src/bootstrap/setup.rs

```diff
@@ -20,7 +21,30 @@ impl Profile {
     }
 
     pub fn all() -> impl Iterator<Item = Self> {
-        [Profile::Compiler, Profile::Codegen, Profile::Library, Profile::User].iter().copied()
+        [Profile::Library, Profile::Compiler, Profile::Codegen, Profile::User].iter().copied()
+    }
+
+    pub fn abbrev_all() -> impl Iterator<Item = (String, Profile)> {
+        ('a'..).map(|c| c.to_string()).zip(Self::all())
```

Comment:
Also, this seems a little special purpose - maybe make it local to `interactive_path`?

Reply:
> Any reason to allocate instead of returning `char`?

Concretely, the callers want to pass this to `format!("...{}...",...,c,...)`.  Formatting a char prints its character code.  I wanted to put the necessary special measures in one place.  It seemed to me that we really don't care about the performance cost of allocating in a command line program's usage handling, so I felt that just making everything be `String` was simplest and most convenient.

Reply:
(This is the same reason I have `Profile::purpose` return a `String` rather than `&'static str`.)

Reply:
> Also, this seems a little special purpose - maybe make it local to `interactive_path`?

That is much better!  Thanks for this suggestion.

Reply:
I'm moving the `a b c` stuff from `Profile as FromStr` into `interactive_path` too, so now those abbrevs are just there.

Reply:
> Formatting a char prints its character code

Ahhh, this is what I was missing. Yeah this is fine then, performance doesn't matter for bootstrap.